lesson21/index.js

```diff
@@ -0,0 +1,245 @@
+    const titleKeys = Object.keys(tableTitlesData);
+    const titleNames = Object.values(tableTitlesData);
+    const tableHead = createTableHead(titleNames);
+    const tableBody = createTableBody(data, titleKeys);
+
+    parent.appendChild(table).appendChild(tableHead).after(tableBody);
+};
+
+
+const createTableHead = titlesData => {
+    const tableHead = document.createElement("thead");
+    const tr = createElementWithClassName("tr", "table__row");
+    const fragment = document.createDocumentFragment();
+    
+    for(let i = 0; i < titlesData.length; i++){
+        const th = createElementWithClassName("th", "table__title js-table-title");
+        th.textContent = titlesData[i];
+        fragment.appendChild(th);
+    }
+    tableHead.appendChild(tr).appendChild(fragment);
+
+    return tableHead;
+};
+
+const createTableBody = (data, keys) => {
+    const tableBody = document.createElement("tbody");
+    const fragment = document.createDocumentFragment();
+
+    for(let i = 0; i < data.length; i++){
+        const tr = createElementWithClassName("tr", "table__row js-table-row");
+        const td = createTableContents(data[i], keys);
+        fragment.appendChild(tr).appendChild(td);
+    }
+    tableBody.appendChild(fragment);
+
+    return tableBody;
+};
+
+const createTableContents = (data, keys) => {
+    const fragment = document.createDocumentFragment();
+
+    for(let i = 0; i < keys.length; i++){
+        const td = createElementWithClassName("td", "table__contents js-table-contents");
+        td.textContent = data[keys[i]];
+        fragment.appendChild(td);
+    }
+
+    return fragment;
+};
+
+const createSortButtons = () => {
+    const ul = createElementWithClassName("ul", "sort-buttons");
+    ul.id = "js-sort-buttons";
+
+    const buttonConfig = [
+        { src: "../img/icon-both.svg", alt: "idを昇順に並び替える", dataSet: "default"},
+        { src: "../img/icon-asc.svg", alt: "idを降順に並び替える", dataSet: "asc"},
+        { src: "../img/icon-desc.svg", alt: "idを順不同に並び替える", dataSet: "desc"}
+    ]
+
+    const fragment = document.createDocumentFragment();
+    buttonConfig.forEach((item, index) => {
+        const li = createElementWithClassName("li", "sort-buttons__item js-sort-button-item");
+        const button = createElementWithClassName("button", "sort-buttons__btn");
+        const img = createElementWithClassName("img", "sort-buttons__img");
+
+        li.dataset.index = index;
+        li.dataset.currentStatus = item.dataSet;
+        img.src = item.src;
+        img.alt = item.alt;
+        fragment.appendChild(li).appendChild(button).appendChild(img);
+    })
+    ul.appendChild(fragment);
+    return ul;
+};
+
+const renderSortButton = () => {
+    document.querySelector(".js-table-title").insertAdjacentElement('beforeend', createSortButtons());
+};
+
+const setButtonForInitDisplay = () => {
+    document.querySelector(".js-sort-button-item").classList.add("is-active");
```

Comment:
IMO
Instead of activating the first element, it would be better to activate the sort button for the default state
```suggestion
    document.querySelector('[data-current-status="default"]').classList.add("is-active");
```

Reply:
Thank you for your review!
That's right!!
I were not making the best use of the data attribute.
I fixed it.
[e9e0666](https://github.com/haru-programming/JavaScript-lessons/pull/26/commits/e9e0666139d9fdc19dfa7babefba1316f3284dbe)